### PR TITLE
Add prettier dependency to @wordpress/eslint-plugin 

### LINKED ISSUE #20027: Error: Cannot find module 'prettier' with @wordpress/eslint-plugin v3.4.0
**Describe the bug**

See this example repository with the minimum setup to replicate the issue:
https://github.com/kasparsd/wordpress-eslint-plugin-prettier

Running `eslint` with `.estlintrc.js` config set to:

```json
{
	"extends": [ "plugin:@wordpress/eslint-plugin/recommended" ]
}
```

and `@wordpress/eslint-plugin` as a development dependency set to version `^3.4.0` produces the following error:

```
Error: Cannot find module 'prettier'
Occurred while linting /Users/kaspars/Projects/wp-scripts-eslint-prettier-bug/src/script.js:1
    at Function.Module._resolveFilename (module.js:548:15)
    at Function.Module._load (module.js:475:25)
    at Module.require (module.js:597:17)
    at require (/Users/kaspars/Projects/wp-scripts-eslint-prettier-bug/node_modules/v8-compile-cache/v8-compile-cache.js:161:20)
    at Program (/Users/kaspars/Projects/wp-scripts-eslint-prettier-bug/node_modules/eslint-plugin-prettier/eslint-plugin-prettier.js:162:26)
    at listeners.(anonymous function).forEach.listener (/Users/kaspars/Projects/wp-scripts-eslint-prettier-bug/node_modules/eslint/lib/linter/safe-emitter.js:45:58)
    at Array.forEach (<anonymous>)
    at Object.emit (/Users/kaspars/Projects/wp-scripts-eslint-prettier-bug/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (/Users/kaspars/Projects/wp-scripts-eslint-prettier-bug/node_modules/eslint/lib/linter/node-event-generator.js:254:26)
    at NodeEventGenerator.applySelectors (/Users/kaspars/Projects/wp-scripts-eslint-prettier-bug/node_modules/eslint/lib/linter/node-event-generator.js:283:22)
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! @kasparsd/wp-scripts-eslint-prettier-bug@1.0.0 lint: `eslint ./src`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the @kasparsd/wp-scripts-eslint-prettier-bug@1.0.0 lint script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/kaspars/.npm/_logs/2020-02-04T10_54_35_609Z-debug.log
```

See the output on Travis:
https://travis-ci.com/kasparsd/wp-scripts-eslint-prettier-bug/jobs/283358536#L519-L538

Downgrading `@wordpress/eslint-plugin` to version `3.3.0` fixes the problem -- see this pull request:

https://github.com/kasparsd/wordpress-eslint-plugin-prettier/pull/1

and the Travis CI output for the change:

https://travis-ci.com/kasparsd/wp-scripts-eslint-prettier-bug/jobs/283361201#L463-L467

**To reproduce**

1. Clone the sample repository `git clone https://github.com/kasparsd/wordpress-eslint-plugin-prettier`
2. Run `npm install`
3. Run `npm run lint`

**Expected behavior**

`eslint` runs as expected.

**Desktop (please complete the following information):**
 - macOS 10.15.3
 - Terminal
 - node v8.17.0, npm 6.13.6.

### PR DESCRIPTION
**Pull request by kasparsd**
## Description

Fixes #20027.

In #19963 we added `eslint-plugin-prettier` as a dependency to `@wordpress/eslint-plugin` in f5c44ea36ca1928604e43c1145845ded5b6993d3 which [requires `prettier` as a peer dependency](https://github.com/prettier/eslint-plugin-prettier/blob/9a47a6feab691cf228d184c103d4cab99b464d0b/package.json#L35).

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR. .

### CODE REVIEW

File: packages/eslint-plugin/package.json [hunk 1]
```diff
@@ -33,6 +33,7 @@
 		"eslint-plugin-react": "^7.14.3",
 		"eslint-plugin-react-hooks": "^1.6.1",
 		"globals": "^12.0.0",
+		"prettier": "npm:wp-prettier@1.19.1",
```
Comment (kasparsd) [inline packages/eslint-plugin/package.json]: This matches what was used in #18048 for `wp-scripts` in d20fa4513194ff1ecab7bf8493162cad50ed2332.

**Review by epiqueras — APPROVED** [review]

**Comment by epiqueras** [comment]
You need to commit lock-file changes.

**Comment by ntwb** [comment]
I think we need to take some time to test the configuration combinations that `@wordpress/scripts` and `@wordpress/eslint-plugin` can be used.

See also https://github.com/WordPress/gutenberg/commit/0a96d2c70efab1d377be018272fc55eaacc33316

**Comment by epiqueras** [comment]
How so?

**Comment by kasparsd** [comment]
> You need to commit lock-file changes.

@epiqueras I'm not seeing a lock file [in the root directory of this package](https://github.com/WordPress/gutenberg/tree/6fc88cade742036261bfb31140ed0bc134e9407d/packages/eslint-plugin) so I wasn't sure it's required. Please confirm.

**Comment by epiqueras** [comment]
It's at the root of the repo (Lerna hoisting). Just run `npm install`.

**Comment by ntwb** [comment]
The _lock_ file needs to b updated using Lerna, not `npm install`

See: https://github.com/WordPress/gutenberg/tree/master/packages#adding-new-dependencies

Though I'm not sure how Lerna will handle the _npm_ alias /shrug
> `lerna add npm:wp-prettier eslint-plugin`

**Comment by epiqueras** [comment]
Both work.

**Comment by ntwb** [comment]
Let's follow the docs please @epiqueras, Lerna has idiosyncratic behaviour that lends to having the lock file out of date, the docs were written to mitigate this scenario as much as possible from past expereinces

**Comment by kasparsd** [comment]
Lock file updated in 2026a6022445bf010d229912c7fc409526068102

**Comment by gziolo** [comment]
It looks good as far as I can tell. In the lock, there  is only one line updated which means the same version is shared 👍 

@kasparsd, thanks for catching and proposing the fix. We didn't test ESLint plugin in isolation :)

**Comment by epiqueras** [comment]
>Let's follow the docs please @epiqueras, Lerna has idiosyncratic behaviour that lends to having the lock file out of date, the docs were written to mitigate this scenario as much as possible from past expereinces

I thought we had a `postinstall` hook. If not, we should have one. We also need to correct that documentation to use the local version of Lerna.

Does that sound good? I can open a PR.

cc @aduth

**Pull request merged**